### ISSUE TITLE
Can't modify response data in afterRemote

### ISSUE THREAD

**Issue by EmileSpecs** [strongloop/loopback#1816]
This is a very strange issue and annoying because it's just one of those that's not supposed to not work!

I have this mixin which is simply supposed to delete the password from the response data just like in the documentation:
```
Model.afterRemote(options.from, function(ctx, modelInstance, next) {
		if (ctx.result) {
			if (Array.isArray(ctx.result)) {
				ctx.result.forEach(function(result) {
					delete result.password;
					result.more = 'Add more!';
				});
			} else {
				delete ctx.result.password;
			}
		}
		console.log('Deleted password', ctx.result);
		next();
	});
```
The result still includes the password and the `more` property that I've added doesn't exist in the response data.

Interestingly if I set `result.password = null;` the console shows the password as null!?
So it won't delete it, it won't add a property but it will let me set it to null. 

Some bug? Caching the results? Any help please, I'm pulling out my hair!

**Comment by jannyHou** [comment]
hi @EmileSpecs I am trying to reproduce your problem, can you send me your model's .json and .js file?

**Comment by EmileSpecs** [comment]
Hi @jannyHou , thanks!

Model:
```
{
  "name": "EmailAccount",
  "base": "PersistedModel",
  "strict": false,
  "idInjection": false,
  "options": {
    "validateUpsert": true
  },
  "mixins": {
    "Timestamp": true,
    "Removepassword": { 
      "from": "find"
    }
  },
  "properties": {
    "archiveConfig": {
      "type": "object"
    },
    "importConfig": {
      "type": "object"
    },
    "tags": {
      "type": "object"
    },
    "user": {
      "type": "string",
      "required": true
    },
    "password": {
      "type": "string"
    },
    "host": {
      "type": "string",
      "required": true
    },
    "archive": {
      "type": "boolean",
      "required": true,
      "default": false
    },
    "import": {
      "type": "boolean",
      "required": true,
      "default": false
    },
    "slug": {
      "type": "string",
      "required": true
    }
  },
  "validations": [],
  "relations": {
    "emails": {
      "type": "hasMany",
      "model": "Postbox",
      "foreignKey": ""
    }
  },
  "acls": [],
  "methods": []
}
```
This is the mixin:
```
module.exports = function(Model, options) {
	options.from = options.from || '**';
	
	Model.afterRemote(options.from, function(ctx, modelInstance, next) {
		if (ctx.result) {
			if (Array.isArray(ctx.result)) {
				ctx.result.forEach(function(result) {
					//delete result.password;
					result.password = null;
				});
			} else {
				delete ctx.result.password;
			}
		}
		console.log('Delete password', ctx.result);
		next();
	});
}
```

**Comment by jannyHou** [comment]
Hi @EmileSpecs, since I have nothing in front-end, I did everything from explorer. What I did is creating some sample emailaccounts and then clicking endpoint GET /emailaccount.
Right the mixin doesn't work, but I am not sure whether my process reproduced your problem in a right way, so I need to confirm with you first then I can move to the next step to fix it.
Thanks.

**Comment by jannyHou** [comment]
@EmileSpecs  well I think I find the solution, hope it's not too late for you :) Please check this issue: 
https://github.com/strongloop/loopback/issues/1162
I tried and it works. The solution is to use `unsetAttribute` instead of `delete`,  so in your case, it should be `result.unsetAttribute('password')`

**Comment by EmileSpecs** [comment]
@jannyHou thanks! I saw that in the documentation but didn't think the resulting array of objects would be instances of the model and thus have that function so I just ignored it.

Also because of the example in the documentation... I suggest that the documentation be updated, since it's obviously not correct.

I'm not sure when the `unsetAttribute` function wouldn't be be available but I'm now using:
```
if (typeof result.unsetAttribute === 'function') result.unsetAttribute('password');
else delete result.password;
```
...just to be sure that the job gets done either way.

Thanks for the help!
